stores/tasks.ts

```diff
@@ -30,64 +33,92 @@ export const useTasksStore = defineStore('tasks', () => {
     return orderedTasks
   })
 
-  watchEffect(() => {
-    if (!isReady.value && tasks.value) {
-      isReady.value = true
-    }
-  })
-
-  // Initialize task priority after first fetch
-  watchEffect(() => {
-    if (permissions.canEditLabels && isReady.value) {
-      initializePriority()
-    }
-  })
-
-  // Merge task-derived columns with existing columns
-  watchEffect(() => {
-    if (tasksByColumn.value) {
-      board.mergeColumns(Object.keys(tasksByColumn.value))
-    }
-  })
-
-  async function initializePriority() {
-    if (!tasks.value) {
-      return
+  const tasksWithoutPriority = computed(() => {
+    if (!permissions.canEditLabels || !tasks.value) {
+      return null
     }
 
     const tasksWithoutPriority: Task[] = []
-    let highestPriority = 0
 
     for (const task of tasks.value) {
       if (task.rpb.priority === null) {
         tasksWithoutPriority.push(task)
-      } else if (task.rpb.priority > highestPriority) {
-        highestPriority = task.rpb.priority
       }
     }
 
-    for (const [index, task] of tasksWithoutPriority.entries()) {
-      const priority = highestPriority + (index + 1) * taskPriorityIncrement
+    return tasksWithoutPriority
+  })
 
-      await updateTaskLabels(task, [...task.labels, createDataLabel('priority', priority)])
-    }
+  const { mutate: initializePriority, isPending: isInitializePriorityPending } = useMutation({
+    async mutationFn() {
+      if (!tasksByColumn.value || !tasksWithoutPriority.value) {
+        return
+      }
 
-    if (tasksWithoutPriority.length > 0) {
-      await refreshTasks()
-    }
-  }
+      function getHighestPriorityInColumn(column: string) {
```

Comment:
I think it may be better and cleaner calculating and keeping that info somewhere else in the store for this and whichever later use with something like (roughly):

```ts
const tasksByColumn: (Column & { highestPriorityTask: <Computed<Task>> })[] = computed(() => /*...*/)
```

The above would self-resolve all the contents and is practically a different projection of the our state + some aggregation results. Notice how I'm suggesting to leverage computed properties for their lazy computation and native caching abilities for performance (beyond the obvious reactivity DX they offer).

Reply:
If we were always storing the tasks sorted by priority then wouldn't `highestPriority` would be obsolete given that it would always be `columnMap[0].priority`? That would perhaps allow us to further rethink other assumptions and pehaps have a cascade of simplifications/renames too.

Having them sorted would also make sense since we're listing them sorted anyway.

What do you think?

Reply:
The tasks are indeed stored sorted by priority (after they've been split into their respective columns). The only issue with getting the top priority from the last task is that the array may also include tasks without priority. However this is easily solved by finding the last task that **with** a priority.
```typescript
tasks[column].findLast(task => task.rpb.priority !== null).rpb.priority
```

This kind of leads us back to the original solution, where I use a function to get the top priority instead of having it in a computed property.

Renamed "highestPriority" to "topPriority" as suggested. (Sidenote: Note sure where you got the `-1`. Priority should always be positive)

P.S. Can't find this comment in the files changed (because the code has been updated) so I can't add it as part of a the review and need to post it standalone 🤦

Reply:
Good on the rollback, I also think it's better that way too.

In the interest of time I'm calling it good enough and resolving, but I'm curious why we're opting for a function instead of a (also suggested) computed property which would be a tad cleaner and more performant due to its cached (and lazily computed! :+1: ) nature.